utils/src/main/scala/za/co/absa/enceladus/utils/explode/ExplodeTools.scala

```diff
@@ -0,0 +1,390 @@
+                               pathTo: Seq[String],
+                               placementCol: Option[String] = None): DataFrame = {
+    def processStruct(schema: StructType, path: Seq[String], parentCol: Option[Column]): Seq[Column] = {
+      val currentField = path.head
+      val isLeaf = path.lengthCompare(1) <= 0
+      var isFound = false
+
+      val newFields = schema.fields.flatMap(field => {
+        if (field.name == columnFrom) {
+          // This removes the original column name (if any) and the transient column
+          Nil
+        } else if (!isFound && isLeaf && placementCol.isDefined && placementCol.get == field.name) {
+          isFound = true
+          Seq(col(s"`$columnFrom`").as(currentField))
+        } else if (!isFound && field.name == currentField) {
+          field.dataType match {
+            case _ if isLeaf =>
+              isFound = true
+              Seq(col(s"`$columnFrom`").as(currentField))
+            case st: StructType =>
+              val newFields = processStruct(st, path.tail, Some(getFullFieldPath(parentCol, field.name)))
+              if (newFields.lengthCompare(1) == 0) {
+                // a struct that can be null
+                val fld = newFields.head
+                Seq(when(fld.isNotNull, struct(newFields: _*)).otherwise(null).as(field.name))
+              } else {
+                // Normat struct
+                Seq(struct(newFields: _*).as(field.name))
+              }
+            case _ =>
+              throw new IllegalArgumentException(s"$currentField is not a struct in ${pathTo.mkString(".")}")
+          }
+        } else {
+          Seq(getFullFieldPath(parentCol, field.name).as(field.name))
+        }
+      })
+      if (!isFound && isLeaf) {
+        val c = col(s"`$columnFrom`")
+        newFields :+ when(c.isNotNull, c).otherwise(null).as(currentField)
+      } else {
+        newFields
+      }
+    }
+
+    df.select(processStruct(df.schema, pathTo, None): _*)
+  }
+
+  def addSuperTransientField(inputDf: DataFrame, arrayColPathName: String): (DataFrame, String) = {
+    val colName = SchemaUtils.getUniqueName(superTransientColumnName, Some(inputDf.schema))
+    val nestedColName = (arrayColPathName.split('.').dropRight(1) :+ colName).mkString(".")
+    val df = DeepArrayTransformations.nestedAddColumn(inputDf, nestedColName, _ => lit(null))
+    (df, nestedColName)
+  }
+
+  private def getFullFieldPath(parentCol: Option[Column], fieldName: String): Column = {
+    parentCol match {
+      case None => col(fieldName)
+      case Some(parent) => parent.getField(fieldName)
+    }
+  }
+
+  private def getRootLevelPrefix(fieldName: String, prefix: String, schema: StructType): String = {
+    getClosestUniqueName(s"${fieldName}_$prefix", schema)
+      .replaceAll("\\.", "_")
+  }
+
+  private def extructFieldFromStruct(df: DataFrame, structFieldName: String): (DataFrame, Column) = {
+    val tmpColName = getUniqueName("tmp_ext", Some(df.schema))
```

Comment:
This method is never used
 \+ I think there is a typo. I think it should be `extractFieldFromStruct`